### ISSUE TITLE
dir() doesn't list instance attributes

### ISSUE THREAD

**Issue by deshipu** [bbcmicrobit/micropython#112]
If you run this code:
```
class A:
    def __init__(self):
        self.x = 3

a = A()
dir(a)
```

You will get:
```
['__qualname__', '__module__', '__init__']
```

while I would expect:
```
['__qualname__', '__module__', '__init__', 'x']
```

**Comment by dpgeorge** [comment]
This should be discussed upstream, ie at https://github.com/micropython/micropython

**Comment by dpgeorge** [comment]
Will be fixed by #165.

**Comment by dpgeorge** [comment]
Fixed.
